client/executor/common/src/runtime_blob/runtime_blob.rs

```diff
@@ -179,8 +171,13 @@ impl RuntimeBlob {
 			return Err(WasmError::Other("memory section is empty".into()))
 		}
 		for memory_ty in memory_section.entries_mut() {
-			let min = memory_ty.limits().initial().saturating_add(extra_heap_pages);
-			let max = memory_ty.limits().maximum().map(|max| std::cmp::max(min, max));
+			let initial = memory_ty.limits().initial();
+			let (min, max) = match heap_pages {
+				HeapPages::Dynamic => (initial, None),
+				HeapPages::Max(max) => (initial, Some(max as _)),
+				HeapPages::ExtraMax(extra) =>
+					(initial + extra as u32, Some(initial + extra as u32)),
+			};
 			*memory_ty = MemoryType::new(min, max);
 		}
```

Comment:
1. We're ignoring and overriding the innate limit set inside of the `.wasm` blob here. Compatibility-wise I think this probably shouldn't be a problem since we never supported any kind of memory growth and none of our runtimes should have the maximum set (IIRC you do need to explicitly pass a flag to the linker to have it set). That said, maybe it'd be worthwhile to keep this door open for the future and support the limit anyway (but allow it to be overridden in emergencies) or explicitly *reject* any runtime which has is set? I'm not entirely sure myself. (cc @pepyakin)
2. Do we need the `Max` variant? As far as I can see it's currently used only in tests.
3. I think a better name for `Dynamic` would maybe be `DynamicUnlimited`. Or better yet, just merge the `Max` variant into `Dynamic` and add a `maximum_pages: Option<u32>` field.
4. We should return an error if the maximum number of pages is less than `initial`.
5. I think a better name for `ExtraMax` would maybe be `Static` and have the extra pages as a named argument?

Reply:
> 2\. Do we need the `Max` variant? As far as I can see it's currently used only in tests.

For PVF execution where we want some fixed maximum.

> 1. We're ignoring and overriding the innate limit set inside of the `.wasm` blob here. Compatibility-wise I think this probably shouldn't be a problem since we never supported any kind of memory growth and none of our runtimes should have the maximum set

Let me throw an error if this is set.